src/Processors/Formats/Impl/PrometheusTextOutputFormat.cpp

```diff
@@ -0,0 +1,104 @@
+#include <optional>
+#include <type_traits>
+#include <Processors/Formats/Impl/PrometheusTextOutputFormat.h>
+#include <Formats/FormatFactory.h>
+#include <Formats/registerWithNamesAndTypes.h>
+#include <IO/WriteHelpers.h>
+#include <DataTypes/DataTypeNullable.h>
+#include <DataTypes/Serializations/ISerialization.h>
+#include <base/defines.h>
+
+namespace DB
+{
+
+constexpr auto FORMAT_NAME = "Prometheus";
+
+static bool isDataTypeString(const DataTypePtr & type)
+{
+    return WhichDataType(type).isStringOrFixedString();
+}
+
+template <typename ResType, typename Pred>
+static void getColumnPos(const Block & header, const String & col_name, Pred pred, ResType & res)
+{
+    static_assert(std::is_same_v<ResType, size_t> || std::is_same_v<ResType, std::optional<size_t>>, "Illegal ResType");
+
+    constexpr bool is_optional = std::is_same_v<ResType, std::optional<size_t>>;
+
+    if (header.has(col_name))
+    {
+        res = header.getPositionByName(col_name);
+        const auto & col = header.getByName(col_name);
+        if (!pred(is_optional ? removeNullable(col.type) : col.type))
+        {
+            throw Exception(ErrorCodes::BAD_ARGUMENTS, "Illegal type '{}' of column '{}' for output format '{}'",
+                col.type->getName(), col_name, FORMAT_NAME);
+        }
+    }
+    else
+    {
+        if constexpr (is_optional)
+            res = std::nullopt;
+        else
+            throw Exception(ErrorCodes::BAD_ARGUMENTS, "Column '{}' is required for output format '{}'", col_name, FORMAT_NAME);
+    }
+}
+
+PrometheusTextOutputFormat::PrometheusTextOutputFormat(
+    WriteBuffer & out_,
+    const Block & header_,
+    const RowOutputFormatParams & params_,
+    const FormatSettings & format_settings_)
+    : IRowOutputFormat(header_, out_, params_), format_settings(format_settings_)
+{
+    const Block & header = getPort(PortKind::Main).getHeader();
+
+    getColumnPos(header, "name", isDataTypeString, pos.name);
```

Comment:
BTW maybe it worth to add some setting to override column names?
I.e. `output_format_prometheus_name_column`
Although maybe it does not worth it, since you can always write alias in the query itself.